src/components/Envoy/EnvoyModal.tsx

```diff
@@ -0,0 +1,234 @@
+import * as API from '../../services/Api';
+import * as AlertUtils from '../../utils/AlertUtils';
+import * as React from 'react';
+import {
+  Button,
+  Card,
+  EmptyState,
+  EmptyStateIcon,
+  EmptyStateVariant,
+  GutterSize,
+  Modal,
+  Spinner,
+  Stack,
+  StackItem,
+  Title,
+  Toolbar,
+  ToolbarGroup,
+  ToolbarItem
+} from '@patternfly/react-core';
+import { Workload } from '../../types/Workload';
+import { Pod } from '../../types/IstioObjects';
+import ToolbarDropdown from '../ToolbarDropdown/ToolbarDropdown';
+import AceEditor from 'react-ace';
+import { aceOptions } from '../../types/IstioConfigDetails';
+import { style } from 'typestyle';
+import { SummaryWriterBuilder } from './writers/BaseWriter';
+
+// Enables the search box for the ACEeditor
+require('ace-builds/src-noconflict/ext-searchbox');
+
+const resources: string[] = ['all', 'bootstrap', 'clusters', 'listeners', 'routes'];
+
+const displayFlex = style({
+  display: 'flex'
+});
+
+const toolbarSpace = style({
+  marginLeft: '1em'
+});
+
+type EnvoyDetailModalProps = {
+  namespace: string;
+  workload: Workload;
+  isOpen: boolean;
+  onClose: (changed?: boolean) => void;
+};
+
+type EnvoyDetailProps = {
+  namespace: string;
+  workload: Workload;
+  onClose: (changed?: boolean) => void;
+};
+
+type EnvoyDetailState = {
+  config: any;
+  resource: string;
+  fetch: boolean;
+  pod: Pod;
+};
+
+export const EnvoyDetailsModal = ({ namespace, workload, isOpen, onClose }: EnvoyDetailModalProps) => (
```

Comment:
Perhaps you may want to reuse here same pattern we are using for the Wizards.

There is a skeleton used called "WorkloadWizard" which basically performs same steps this is doing:
- Passing the workload, namespace, and which sub-wizard you want to open.
- Taking care of common controls: Create/Cancel buttons, title, etc.

So, all EnvoyDetail class would suit inside the WorkloadWizard and leave as it is, but then we can share and modify aspect of the modal for same category (Workload, Services, etc).

Reply:
I don't know if I fully understand this one.

This is not exactly a wizard, this is a regular modal, isn't it?. Perhaps, the modal you use in the `WorkloadWizard` could be extracted to a common component that everyone could use. 

On the other hand, I don't know why we should use a `sub-wizard` prop in this case. There is a drowpdown with multiple values, but they are not coming from the outside component neither the initial selection.

Regarding the common controls, this scenario doesn't need any "Create"/"Update", it is just informative actions you can do in this modal. There isn't any impact in the backend after finishing the task in the modal.

One thing I could do to fit more the "pattern" is not to have two components one for the modal and another for the content of the modal.